packages/flame_studio/lib/src/widgets/panels/hierarchy_view.dart

```diff
@@ -0,0 +1,221 @@
+      hasChildren: hasChildren,
+      isExpanded: isExpanded,
+      isFirst: out.isEmpty,
+      isLast: depth == 0 && !(isExpanded && hasChildren),
+    );
+    if (hasChildren) {
+      expanderIcon = MouseRegion(
+        cursor: SystemMouseCursors.click,
+        child: GestureDetector(
+          onTap: () => _toggle(node.component),
+          child: expanderIcon,
+        ),
+      );
+    }
+    final indent = SizedBox(width: 15.0 * depth);
+    final componentName = MouseRegion(
+      cursor: SystemMouseCursors.click,
+      child: GestureDetector(
+        onTap: () => _toggleOrSelect(node.component),
+        child: Container(
+          padding: const EdgeInsets.symmetric(horizontal: 4, vertical: 2),
+          decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(4.0),
+            color: isSelected ? const Color(0xffc78938) : null,
+          ),
+          child: Text(
+            node.name,
+            style: const TextStyle(
+              color: Color(0xffffffff),
+            ),
+          ),
+        ),
+      ),
+    );
+    final punctuation = Transform(
+      transform: Matrix4.translationValues(-3.0, 0, 0),
+      child: Text(
+        hasChildren ? (isExpanded ? ' {' : ' {...}') : ',',
+        style: const TextStyle(color: Color(0x66f5d49a)),
+      ),
+    );
+
+    out.add(
+      Row(
+        children: [
+          expanderIcon,
+          indent,
+          componentName,
+          punctuation,
+        ],
+      ),
+    );
+    if (isExpanded && hasChildren) {
+      for (final childNode in node.children!) {
+        _buildList(childNode, depth + 1, out);
+      }
+      out.add(
+        Row(
+          children: [
+            _ExpanderIcon(hasChildren: false, isLast: depth == 0),
+            SizedBox(width: 15.0 * depth + 5.0),
+            const Text(
+              '}',
+              style: TextStyle(color: Color(0x66f5d49a)),
+            ),
+          ],
+        ),
+      );
+    }
+  }
+
+  /// Used by the expander icons in the gutter: the component expands/collapses
+  /// without affecting the selection state.
+  void _toggle(Component component) {
+    setState(() {
+      if (expandedComponents.contains(component)) {
+        expandedComponents.remove(component);
+      } else {
+        expandedComponents.add(component);
+      }
+    });
+  }
+
+  void _toggleOrSelect(Component component) {
+    final isExpanded = expandedComponents.contains(component);
+    final isSelected = selectedComponent == component;
+    setState(() {
+      if (isExpanded && isSelected) {
+        expandedComponents.remove(component);
+      } else {
+        expandedComponents.add(component);
+      }
+      selectedComponent = component;
+    });
+  }
+}
+
+class _ExpanderIcon extends StatelessWidget {
+  const _ExpanderIcon({
+    required this.hasChildren,
+    this.isExpanded = false,
+    this.isFirst = false,
+    this.isLast = false,
+  });
+  final bool hasChildren;
```

Comment:
```suggestion

  final bool hasChildren;
```

Reply:
This is something with GitHub's display of whitespace, the code itself looks fine (and the formatter would have complained if it wasn't the case).

Reply:
I don't think there is a rule for this in the formatter? It looks like the same if I look at the raw file.

Reply:
done